GetIntoTeachingApi/Models/Validators/TeachingEventValidator.cs

```diff
@@ -1,18 +1,32 @@
 ﻿using System;
 using FluentValidation;
 using FluentValidation.Validators;
+using GetIntoTeachingApi.Services;
 
 namespace GetIntoTeachingApi.Models.Validators
 {
     public class TeachingEventValidator : AbstractValidator<TeachingEvent>
     {
-        public TeachingEventValidator()
+        private readonly ICrmService _crm;
+
+        public TeachingEventValidator(ICrmService crm)
         {
-            RuleFor(teachingEvent => teachingEvent.ReadableId).NotEmpty();
+            _crm = crm;
+
+            RuleFor(teachingEvent => teachingEvent.ReadableId)
+                .NotEmpty()
+                .Must(IsPartialUrlUnique)
+                .When(teachingEvent => teachingEvent.Id == null)
+                .WithMessage("External event title must be unique");
```

Comment:
The field name will be part of the message, so I think this would read better just as `Must be unique`